### PR TITLE
Cleanup Java Network.getLocalAddress

### LINKED ISSUE #2841: Revisit Java Network.getLocalAddress
I was looking why we have to keep the include loopback in Java unlike in other mappings.

Seems we only set this flag to false in https://github.com/zeroc-ice/ice/blob/55308b51034a21c7f5ccf3c8c1ef830c3476c444/java/src/Ice/src/main/java/com/zeroc/Ice/Network.java#L565

This overload is only used by the syslog logger to get the localhost address

https://github.com/zeroc-ice/ice/blob/55308b51034a21c7f5ccf3c8c1ef830c3476c444/java/src/Ice/src/main/java/com/zeroc/Ice/SysLoggerI.java#L75

It would make more sense to use a property to configure the syslog daemon, with a default. Then we can get rid of this  getLocalAddress overload and the include loopback flag.

For a separate issue

_Originally posted by @pepone in https://github.com/zeroc-ice/ice/pull/2839#discussion_r1785109797_

### PR DESCRIPTION
**Pull request by pepone**
Fix #2841

### CODE REVIEW

**Review by bernardnormier — APPROVED** [review]
Looks good.

File: java/src/Ice/src/main/java/com/zeroc/Ice/SysLoggerI.java [hunk 1]
```diff
@@ -10,69 +10,18 @@
...
-    }
-
-    private SysLoggerI(String prefix, int facility) {
-        initialize(prefix, facility);
+    public SysLoggerI(String prefix, String facilityString, String host, int port) {
+        this(prefix, parseFacility(facilityString), parseHost(host), port);
     }
 
-    private void initialize(String prefix, int facility) {
+    private SysLoggerI(String prefix, int facilityString, InetAddress host, int port) {
```
Comment (InsertCreativityHere) [inline java/src/Ice/src/main/java/com/zeroc/Ice/SysLoggerI.java:17]: `int facilityString` looks dubious to me here.
What is this variable representing?

**Review by InsertCreativityHere — APPROVED** [review]
Looks good.

File: java/src/Ice/src/main/java/com/zeroc/Ice/SysLoggerI.java (hunk 1 above)
Comment (pepone) [reply java/src/Ice/src/main/java/com/zeroc/Ice/SysLoggerI.java]: Renamed to facility, it is correspond to the syslog numerical facility.

**Pull request merged**